client/src/App.jsx

```diff
@@ -1,20 +1,180 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import axios from 'axios';
+// import dummy from '../../dummydata.js';
+import moment from 'moment';
+import Search from './components/Search.jsx';
+import './main.css';
+
+
 
 class Questions extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      test: 'What are you doing here?'
+      product: {},
+      searchRequest: false
+    }
+    this.changeVote = this.changeVote.bind(this);
+  }
+
+  componentDidMount() {
+    let id = window.location.pathname
+    if (id !== '/') {
+      axios.get(`/products${id}`)
+        .then(response => {
+          // console.log(response, `this is is going well`)
+          // console.log(response, `DATA`)
+          this.setState({ product: response.data });
+        })
+        .catch(err => {
+          console.log(err)
+        });
     }
   }
 
+  changeVote(event) {
+    event.preventDefault()
+    const product_id = event.target.elements.product_id.value;
+    const question_id = event.target.elements.question_id.value;
+    const voteValue = event.target.elements.voteValue.value;
+
+    // makes POST request to update the question's vote count
+    axios.post(`/ask/vote/question/${question_id}`,
+      {
+        vote: voteValue,
+        product: product_id
+      })
+      .then((response) => {
+        console.log(response,  `VOTES RESPONSE FROM SERVER`);
+        const questionId = response.data.question_id;
+        const voteValue = response.data.votes;
+        const questions = [...this.state.product.questions];
+        // find the question_id and update the vote value
+        questions.forEach(question => {
+          if (question.question_id === questionId) {
+            question.votes = voteValue;
+          }
+        });
+        this.setState({ [questions]: questions });
+      });
+  }
+
+  searchQueryAndQuestions(searchResult) {
+    // if this function is inbo
+  }
 
   render() {
+    const { product } = this.state;
+    // questios array
+    const data = this.state.product.questions;
     return (
       <div>
-        <h1>Questions will go here</h1>
-        <div>{this.state.test}</div>
+        <div id="ask-btf_feature_div">
```

Comment:
Have you thought about making wrappers that contain these divs and return their child components? Seems like there are a lot of divs in the App component. Usually, you want App to just contain React components and you let those components handle what divs they need.

Reply:
I'm not too sure on how to use wrappers :( but thanks tho.